### PR TITLE
Add basic tests for ProseMirror menu buttons.

### PR DESCRIPTION
**Pull request by tadhg-ohiggins**
If all the buttons
Are clicked in the deep forest
Will anyone know?

Adds a basic check to verify that the ProseMirror editor menu bar has the buttons we think it should have.

### CODE REVIEW

File: api/document/js/__tests__/menu.test.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { flatten } from 'lodash/array';
+import { EditorView } from 'prosemirror-view';
+import schema, { factory } from '../schema';
+import { EditorState } from 'prosemirror-state';
+import menu from '../menu';
+import { Api, JsonApi } from '../Api';
+
+describe('menu functions', () => {
+  it('has the expected buttons', () => {
```
Comment (toolness) [inline api/document/js/__tests__/menu.test.ts]: Should this be "_have_ the expected buttons"?  I'm just trying to read out the first args of `describe()` and `it()` in a sentence, and it comes out to "menu functions has the expected buttons" which sounds a bit funky.

Comment (cmc333333) [reply api/document/js/__tests__/menu.test.ts]: Maybe s/menu functions/menu function? `it('have ` also isn't quite right...

Comment (tadhg-ohiggins) [reply api/document/js/__tests__/menu.test.ts]: This is an amusing set of comments to find… I was reading the “sentences” purely as “it…” and not as including the argument to `describe`. I think the easiest change to make here is to `s/menu functions/menu`.

**Comment by tadhg-ohiggins** [comment]
Thanks @cmc333333 for helping me find my way through the ProseMirror woods to the relevant menu details.

**Pull request merged**